packages/framework/src/types/config.types.ts

```diff
@@ -23,4 +23,10 @@ export type ClientOptions = {
    * Defaults to true.
    */
   strictAuthentication?: boolean;
+
+  /**
+   * Additional fetch options to be passed to the fetch API.
+   * This can be used to set custom agent, headers, etc.
+   */
+  fetchOptions?: Record<string, any>;
```

Comment:
To provide better DX, I suggest typing this as a [RequestInit](https://developer.mozilla.org/en-US/docs/Web/API/RequestInit).

Reply:
@wh5938316 any updates on this?